lib/rubocop/cop/betterment/unscoped_find.rb

```diff
@@ -49,11 +50,20 @@ def on_send(node)
                 static_method_name(node.method_name)
             ) && !@unauthenticated_models.include?(Utils::Parser.get_root_token(node))
 
-          add_offense(node) if find_param_arg(arg_nodes)
+          add_offense(node) if find_param_arg(arg_nodes) || graphql_namespace?(node)
         end
 
         private
 
+        def graphql_namespace?(node)
+          return true if processed_source.path&.match?(GRAPHQL_PATTERN)
+
+          node
+            .ancestors
+            .select { |n| n.class_type? || n.module_type? }
+            .any? { |n| n.identifier.to_s.match?(GRAPHQL_PATTERN) }
+        end
```

Comment:
__suggestion:__ This seems to be mixing concerns a bit, since it matches a file path against a pattern and then proceeds to match the stringified representation of the AST against the regexp. It works, but I would suggest splitting it out so it's more obvious.

Another suggestion would be to use RuboCop's built-in matching syntax to match the nodes rather than doing all of it by hand. The code uses some the matching syntax already so it's not much of a stretch, eg:

```ruby
# Add under the two `def_node_matcher` calls, returns only nodes matching the regexp:

def_node_search :find_namespace_nodes, <<~PATTERN, name: GRAPHQL_PATTERN
  (const _ %name)
PATTERN


# At the end of on_send:

add_offense(node) if find_param_arg(arg_nodes) || graphql_file? || graphql_namespace?(node)

# ...

def graphql_file?
  processed_source.path&.match?(GRAPHQL_PATTERN)
end

def graphql_namespace?(node)
  node
    .each_ancestor(:class, :module)
    .any? { |ancestor| find_namespace_nodes(ancestor).any? }
end
```

Reply:
Very nice! This works well, and it's nice to use the intended tools for this. I also tried for a while to see if even the syntax you have here could be more concise using a smarter matcher instead of a full search, but ended up coming up with nothing.

Reply:
(also FYI, I changed your method name to `find_graphql_namespace_nodes` since I felt like there was context missing from it)